packages/node/src/nodeClient.ts

```diff
@@ -46,35 +47,50 @@ export class NodeClient implements Client<Options> {
     // Check if there's 0 events, flush is not needed.
     const arrayLength = this._events.length;
     if (arrayLength === 0) {
-      return { status: Status.Success, statusCode: 200 };
+      return SKIPPED_RESPONSE;
+    }
+
+    // Reset the response listeners and pull them out.
+    const responseListeners = this._responseListeners;
+    this._responseListeners = [];
+
+    try {
+      const eventsToSend = this._events.splice(0, arrayLength);
```

Comment:
Since we are splicing we are mutating the original events array. if this try catch errors, (perhaps bc we could not send events) do we not want to preserve the list of events so that they are not lost?

Reply:
There's actually a large layer of retry mechanisms beneath this clients that handles the what if an event fails :) so we're not very concerned here. I might change this to just replace with a new array instead of mutating though.